src/core/internal/object/lifetime.d

```diff
@@ -0,0 +1,752 @@
+/**
+ * Contains utilities for managing lifetimes.
+ *
+ * This module is not intended to be imported or used directly.  It is
```

Comment:
If I make them `package` then they won't be visible outside of `core.internal.object`, right?  If that's the case, then I won't be able to publicly import them into `object.d`.

Reply:
This goes along with my [original idea](https://github.com/dlang/druntime/pull/2222#issuecomment-399092667) in #2222 to put them all in the root, rename `object.d` to `package.d` and then modify the compiler to look for `package.d` (or the root of the runtime) instead of `object.d`  There were concerns about not being able to use `lifetime.d` in one's own file hierarchy.  But, honestly, I don't undersand that, because they would be under a different folder and users should never import the runtime's `lifetime.d` anyway.

Reply:
It's possible to specify a symbol along with package, i.e. `package(core)` which would make it accessible to the whole `core` package and all its sub packages. But since `object` lives outside any package, that might not work anyway.